### ISSUE TITLE
Remove `hls-haddock-comments-plugin`

### ISSUE THREAD

**Issue by michaelpj** [haskell/haskell-language-server#3702]
It hasn't been updated for the last 3 major GHC versions, I think it's dead.

cc @berberman @kokobd as the listed CODEOWNERS

**Comment by kokobd** [comment]
I agree. I tried, but it seemed that handling comments in exact printing of ghc > 9 is much harder than GHC 8.10.7, and there were too many corner cases. Not sure if the situation has improved since then.

**Comment by fendor** [comment]
@kokobd Hmpf, did you collaborate with @alanz about this? Maybe we can work something out to make it easier to use for library authors?

**Comment by alanz** [comment]
Version 1.7.0.1 of ghc-exactprint correctly prints haddock comments, BUT uses the ones that are not embedded as HsDocTy etc.

Each haddock comment appears as a normal comment, as well as the ones allocated by haddock.  The haddock allocations is not perfect (or if the source is incorrect, more to the point), so not all comments show up in the haddock-specific areas.

So ghc-exactprint now treats them as just comments, and does not exact print the haddock-specific wrappers for them

**Comment by kokobd** [comment]
I'll give the latest version of ghc-exactprint a try!

**Comment by kokobd** [comment]
Played with this again, and I realized the problem is not with "inserting" comments, but with determining if haddock comments are already attached to some AST node.

Look at this example.
```hs
f :: Int
  -- | comment 1
  -> -- | comment 2
  Double -- ^ comment 3
```

If we put any of the three comments there, haddock will generate doc for the parameter `Double`. However, in our code, they are three different cases to check. And I remeber in the case of record fields, the problem is more complex (before and after comma, some other edge cases).

I tried to find the related code in Haddock and reuse that, but failed. Personally I don't feel this plugin worth that much effort, so I gave up.
